src/store/helpers.js

```diff
@@ -26,3 +28,111 @@ export function onlyHandleAPIError (error, handleFn) {
     throw error
   }
 }
+
+/**
+ * Defines a vues module that can be used to handle async request metadata.
+ *
+ * Instantiation:
+
+ const meta = defineRequestModule()
+ export const modules = { meta }
+
+ * Usage:
+
+async save ({ commit, dispatch }, group) {
+  dispatch('meta/request', {
+    id: `save-${group.id}`,
+    async run () {
+      const updatedGroup = await groups.save(group)
+      commit(types.RECEIVE_GROUP, { group: updatedGroup }) // other commits when successful
+    },
+  })
+},
+ */
+export function defineRequestModule () {
+  const types = {
+    REQUEST: 'Request',
+    RECEIVE_SUCCESS: 'Receive Success',
+    RECEIVE_ERROR: 'Receive Error',
+    CLEAR: 'Clear',
+  }
+
+  const state = {
+    entries: {},
+  }
+
+  const getters = {
+    get: state => id => state.entries[id] || ({
```

Comment:
I am not a fan of the default value being an object, I think we should move away from the convention as I think it is counter intuitive as you cannot check the result for truthiness, but have to look inside for a particular property

Reply:
It needs to be handled on some level, I'd prefer not having to handle it inside the components (means they should be able to expect `isWaiting`, `error` and `success` properties to exist)

Reply:
It could be handled in the parent `groups` module though, via additional getter.